File: src/my_project/easy_problems/from251to300/longest_common_prefix.py
from typing import List 

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        result = ""
        i = 0
        len_s = len(strs)
        
        if len_s == 0:
            return result 
        
        first_word = min(strs, key=len)
        bench_mark = len(first_word)
        
        while i < bench_mark:
            for word in strs:
                print(word, first_word)
                if first_word[i] != word[i]:
                    return result 
            result += first_word[i]
            i+=1
                
        return result
        
        
solution   = Solution()

print(solution.longestCommonPrefix(strs = ["dog","racecar","car"]))

print(min(["flower","flow","flight"], key=len))

print(min( ["dog","racecar","car"], key=len))

File: my_project_typescript/easy_problems_typescript/from1to50/longestCommonPrefix.ts
function longestCommonPrefix(strs: string[]): string {
    let result: string = "";

    if (strs.length === 0){
        return result;
    }
    let count: number = 0;
    for (let letter of strs[0]){
        for (let item of strs){
            if (!item.slice(count, count+ 1).includes(letter)){
                return result;
            }
        }
        result += letter;
        count += 1;
    }
    return result;
};

console.log(longestCommonPrefix(["flower","flow","flight"]));
console.log(longestCommonPrefix(["dog"]));
console.log(longestCommonPrefix(["aa","ab"]));
console.log(longestCommonPrefix(["c","acc","ccc"]));


